### PR TITLE
Fix random fail of `infer_async_wait_for_return_busy` by adding callback with waiting

### PR DESCRIPTION
**Pull request by vurusovs**
### Tickets:
 - CVS-136569

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: src/bindings/c/tests/ov_infer_request_test.cpp [hunk 1]
```diff
@@ -353,6 +353,12 @@ TEST_P(ov_infer_request_test, infer_async_wait_for) {
 TEST_P(ov_infer_request_test, infer_async_wait_for_return_busy) {
     OV_EXPECT_OK(ov_infer_request_set_input_tensor_by_index(infer_request, 0, input_tensor));
 
+    ov_callback_t callback;
+    callback.callback_func = [](void* args) {
+        sleep(3);
```
Comment (riverlijunjie) [inline src/bindings/c/tests/ov_infer_request_test.cpp:358]: In Windows, it should be `Sleep(3)`

**Comment by riverlijunjie** [comment]
@vurusovs  I put this PR into https://github.com/openvinotoolkit/openvino/pull/23973 to solve another timeout issue together, please have a review.

File: src/bindings/c/tests/ov_infer_request_test.cpp (hunk 1 above)
Comment (praasz) [reply src/bindings/c/tests/ov_infer_request_test.cpp:358]: Maybe use standard sleep `std::this_thread::sleep_for` ?

**Comment by vurusovs** [comment]
Closing PR in favor to https://github.com/openvinotoolkit/openvino/pull/23973

**Pull request closed without merging**